src/libs/components/ClusterfunGameComponent.tsx

```diff
@@ -25,22 +25,40 @@ export interface ClusterFunGameProps {
     serverCall: <T>(url: string, payload: any) => Promise<T>
 }
 
+interface ClusterFunComponentState {
+    appModel?: BaseGameModel;
+    UI: React.ComponentType<{ appModel?: any, uiProperties: UIProperties }>
+}
+
 class DummyComponent extends React.Component<{ appModel?: any, uiProperties: UIProperties }>
 {
     render() { return <div>DUMDUMDUM</div>}
 }
 
+// Finalization registry to shut down models when the corresponding component is destroyed.
+// NOTE: This is kind of a hack
+const modelShutdownRegistry = new FinalizationRegistry((model: BaseGameModel) => {
+    if (model.gameState !== GeneralGameState.Destroyed) {
+        Logger.info("Disposing of a destroyed model");
+    } else {
+        Logger.warn("Disposing of a model that was not destroyed");
+    }
+    model.shutdown();
+})
 
 // -------------------------------------------------------------------
 // Main Game Page
 // -------------------------------------------------------------------
 @observer
 export class ClusterfunGameComponent 
-extends React.Component<ClusterFunGameProps>
+extends React.Component<ClusterFunGameProps, ClusterFunComponentState>
 {
-
-    appModel?: BaseGameModel;
-    UI: React.ComponentType<{ appModel?: any, uiProperties: UIProperties }> = DummyComponent
+    constructor(props: ClusterFunGameProps) {
+        super(props);
+        this.state = {
```

Comment:
I've been trying to avoid direct manipulation of state and instead create a local mobx state class.   (Mostly because I find state manipulation tedious, but also because mobx auto-bundles state manipulate when we use the action function).    Making a whole class is probably overkill now that I think about it.   We could probably do something like this:

```
MyComponent ... 
{
    myState = { foo: "ha", bar: 22 }

    constructor() {
        makeAutoObservable(this.myState)
    }
```
Thoughts?